azure/services/inboundnatrules/spec.go

```diff
@@ -51,8 +51,8 @@ func (s *InboundNatSpec) OwnerResourceName() string {
 // Parameters returns the parameters for the inbound NAT rule.
 func (s *InboundNatSpec) Parameters(ctx context.Context, existing interface{}) (parameters interface{}, err error) {
```

Comment:
It could be out of scope for this PR, but it may be worth adding a test for Parameters in `spec_test.go` like in other services: https://github.com/willie-yao/cluster-api-provider-azure/blob/main/azure/services/agentpools/spec_test.go#L149

Reply:
I think that's a good idea. I'll try to add it to this PR.

Reply:
@willie-yao thanks for the nudge, I added `TestParameters()` in `spec_test.go`. The logic here is pretty simple compared to your agentpools example. 😄